### ISSUE TITLE
Per-file options not applied to directories

### ISSUE THREAD

**Issue by groszdaniel** [mpv-player/mpv#11150]
In a command line like this:
`mpv --{ --opt a/ b.mp4 --} c.mp4`
when mpv encounters `a/`, it adds the files in the directory to the playlist. I'd expect that the per-file (file-local) option --opt would be applied to the files in a/ as well as b.mp4 but not c.mp4; however, it's only applied to b.mp4, and not to the files in the directory.

### Important Information

Provide following Information:
- mpv version: 0.35+git.20221222.657fd280
- Linux Distribution and Version: openSUSE Tumbleweed 20230108
- Source of the mpv binary: Main Repository (OSS)
- Window Manager and version: kwin_x11 5.26.5
- GPU driver and version: xf86-video-intel 2.99.917+git8674.25c9a2fcc-lp152.3.7

### Reproduction steps

mpv --no-config --{ --vf=lavfi=[hue=h=90] movies_gd/ --}

### Expected behavior

The filter is applied to the videos in the directory (the colors look funny).

### Actual behavior

The filter is not applied.

### Log file

[mpv_per_file.txt](https://github.com/mpv-player/mpv/files/10419708/mpv_per_file.txt)

**Comment by llyyr** [comment]
Use `mpv --no-config --{ --vf=lavfi=[hue=h=90] movies_gd/* --}`

**Comment by groszdaniel** [comment]
> Use `mpv --no-config --{ --vf=lavfi=[hue=h=90] movies_gd/* --}`

Yes, I get that, but it would be natural if it worked with specifying just the directory too, given that mpv already supports specifying directories on the command line. And with `dir/*` it will still be broken if the directory contains subdirectories.

**Comment by groszdaniel** [comment]
It just occurred me to try: the issue also comes up if, instead of a directory, I specify a zip file. In that case, too, mpv plays the media files contained in the archive, but without any per-file option applied.
